integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/util/HttpsClientRequest.java

```diff
@@ -61,11 +61,14 @@ public static HttpResponse doGet(String requestUrl, Map<String, String> headers)
                 }
                 httpResponse = new HttpResponse(sb.toString(), conn.getResponseCode());
             } catch (IOException ex) {
-                rd = new BufferedReader(new InputStreamReader(conn.getErrorStream()
-                        , Charset.defaultCharset()));
-                String line;
-                while ((line = rd.readLine()) != null) {
-                    sb.append(line);
+                InputStream errorStream = conn.getErrorStream();
```

Comment:
Can't we move this into the try block after checking the conn response code? So that we could handle the streams better

Reply:
In the original code, it gets the error stream if there are any IO exception occurred.

Reply:
We will anyway have to change this method later. Will Have to check conn.getResponseCode() before calling conn.getInputStream() or conn.getErrorStream().